modules/apps/frontend-data-set/frontend-data-set-impl/src/main/java/com/liferay/frontend/data/set/internal/SystemFDSEntryRegistryImpl.java

```diff
@@ -34,125 +29,63 @@ public class SystemFDSEntryRegistryImpl implements SystemFDSEntryRegistry {
 
 	@Override
 	public Map<String, SystemFDSEntry> getSystemFDSEntries() {
-		return _systemFDSEntries.get();
+		if (!_opened) {
+			getSystemFDSEntry(StringPool.BLANK);
+
+			_opened = true;
+		}
+
+		return _systemFDSEntries;
```

Comment:
Also, I know we do this in a lot of places, but returning the map directly allows to modify it from outside, which can lead to unexpected results.

To avoid that you could theoretically wrap it in an unmodifiable map, but I believe that will be removed when merged upstream so may it's worthless doing it.

In any case, if you could leave just the `getSystemFDSEntry(String fdsName)` method and remove this one, you wouldn't have to worry about what I'm saying. So maybe...

Reply:
Unfortunately, we need the method returning the map because the map uses fds names as keys, which end up being the data set object entry ERC. Also, we need to list all the entries in the UI, so accessing the map entries by key is not enough because keys are not known in advance.

Unless we provide a static list of such keys. But that makes things much harder to maintain.

Reply:
As for the modifiablility of the map, yes, perhaps we shall return a copy of it as we don't want any modification to be made from outside this class

Reply:
>Unfortunately, we need the method returning the map because the map uses fds names as keys, which end up being the data set object entry ERC. 

You could also return a `Collection<String>` with key names. That wouldn't expose so much surface (the whole `Map` interface) while letting users do what they need to do.